exercise/reveal_states.py

```diff
@@ -167,6 +169,8 @@ def get_latest_deadline(self) -> Optional[datetime.datetime]:
             deadlines.append(
                 self.max_deviation.get_new_deadline(exercise_dict[self.max_deviation.exercise_id].closing_time)
             )
+        if len(deadlines) == 0:
```

Comment:
If the module contains no exercises, then the `deadlines` list is empty. But couldn't you check that there are no exercises already at the start of the method? It would skip most of the function since the code is useless in that case.